clar2wasm/tests/wasm-generation/values.rs

```diff
@@ -58,23 +58,53 @@ proptest! {
 #[cfg(not(feature = "test-clarity-v1"))]
 mod clarity_v2_v3 {
     use clar2wasm::tools::TestEnvironment;
+    use clarity::vm::types::{BuffData, SequenceData};
     use clarity::vm::Value;
 
     use super::*;
-    use crate::runtime_config;
 
     proptest! {
-        #![proptest_config(runtime_config())]
-
         #[test]
-        fn value_serialized_and_deserialized(val in PropValue::any().prop_filter("Filter condition description", |val| {
+        fn serialize_deserialize_any_value(val in PropValue::any()) {
+            let snippet = format!("(to-consensus-buff? {val})");
+            const INPUT_TYPE_ERROR: &str = "could not determine the input type for the serialization function";
+
+            // Pre-check to discard test cases where `to-consensus-buff?` evaluation
+            // could not determine the type of the input parameter.
+            // For instance, `(to-consensus-buff? (list))` where a `NoType` should not be evaluated.
             let mut env = TestEnvironment::default();
-            env.evaluate(&format!("(to-consensus-buff? {val})")).is_ok()
-        })) {
+            let pre_check = env.evaluate(&snippet);
+            prop_assume!(match pre_check {
+                Ok(_) => true,
+                Err(ref e) if e.to_string().contains(INPUT_TYPE_ERROR) => false,
+                _ => true
+            });
+
+            // Serialize the PropValue to check it against
+            // the `to-consensus-buff?` implementation.
+            let serialized_value = Value::Sequence(SequenceData::Buffer(BuffData {
+                data: Value::from(val.clone()).serialize_to_vec().unwrap()
+            }));
+
+            let res = pre_check.unwrap(); // Safe to unwrap because of the prop_assume!
+            let expected = if res.is_none() {
+                Ok(Some(Value::none()))
+            } else {
+                Ok(Some(Value::some(serialized_value.clone()).unwrap()))
+            };
+
+            // Crosscheck serialization
+            crosscheck(&snippet, expected);
+
+            // Crosscheck deserialization with `from-consensus-buff?`
             crosscheck(
-                &format!("(from-consensus-buff? {} (unwrap-panic (to-consensus-buff? {})))", val.type_string() ,val),
-                Ok(Some(Value::some(val.into()).unwrap()))
-            )
+                &format!(
+                    "(from-consensus-buff? {} {})",
+                    val.type_string(),
+                    serialized_value
+                ),
+                Ok(Some(Value::some(Value::from(val)).unwrap()))
+            );
```

Comment:
If the serialization result is `None`, normally this test shouldn't be called.
So for me there are two problems here:

1. This is called anytime, nothing checks that the serialization succeeded.
2. If we have a `None`, this crosscheck shouldn't be called, so we will have fewer tests for deserialization than serialization. (This is why I advocate for two tests, so that we are sure we run the required number of tests for both functions).

Reply:
Changed to use a different test to `from-consensus-buff?`.